src/main/java/com/ncc/aif/ValidateAIF.java

```diff
@@ -273,6 +316,40 @@ public Resource validateKBAndReturnReport(Model dataToBeValidated) {
      * @return a validation report from which more information can be derived, or null if validation didn't complete
      */
     public Resource validateKBAndReturnReport(Model dataToBeValidated, Model union) {
+        Set<Resource> reports = validateKBAndReturnMultipleReports(dataToBeValidated, union);
+        if (reports.isEmpty()) {
+            return null;
+        } else if (executor != null) {
+            Resource masterReport = null;
+            for (Resource report : reports) {
+                if (masterReport == null) {
+                    masterReport = report;
+                } else {
+                    StmtIterator it = report.listProperties(SH.result);
+                    Model masterModel = masterReport.getModel();
+                    masterModel.add(report.getModel());
+                    while(it.hasNext()) {
+                        masterReport.addProperty(SH.result, it.next().getObject());
+                    }
+                    masterModel.removeAll(report, null, null);
+                }
+            }
+            return masterReport;
+        } else {
+            return reports.iterator().next();
+        }
+    }
+
+    /**
+     * Validate the specified KB and return a validation report.
+     *
+     * @param dataToBeValidated KB to be validated
+     * @param union             unified KB if not null
+     * @return a validation report from which more information can be derived, or null if validation didn't complete
+     */
+    public Set<Resource> validateKBAndReturnMultipleReports(Model dataToBeValidated, Model union) {
```

Comment:
It's not clear when you can/should call `validateKBAndReturnReport` and when you can/should call `validateKBAndReturnMultipleReports`...and why you should want to call one instead of the other.  It seems like `validateKBAndReturnReport` is trying to combine the error reports, which seems like a good goal ultimately...but maybe that only works for single-threaded validations?

Reply:
Both APIs should work for single or multiple threads. I'll add something to the documentation indicating which should be used when.

Reply:
Fixed